src/algebra/homology/local_cohomology.lean

```diff
@@ -186,4 +194,94 @@ begin
        ... ≤ J           : hk,
 end
 
+/-- The diagram of powers of `J` is initial in the diagram of all ideals with
+radical containing `J`. This uses noetherianness. -/
+/- Can this proof be golfed? -/
+instance ideal_powers_initial [hR : is_noetherian R R] :
+  functor.initial (ideal_powers_to_self_le_radical J) :=
+{ out := λ J', begin
+  apply @zigzag_is_connected _ _ _,
+  { intros j1 j2,
+    apply relation.refl_trans_gen.single,
+    -- The inclusions `J^n1 ≤ J'` and `J^n2 ≤ J'` always form a triangle, based on
+    -- which exponent is larger.
+    cases le_total (unop j1.left) (unop j2.left) with h,
+      right, exact ⟨costructured_arrow.hom_mk (hom_of_le h).op (of_as_true trivial)⟩,
+      left, exact ⟨costructured_arrow.hom_mk (hom_of_le h).op (of_as_true trivial)⟩ },
+  { obtain ⟨k, hk⟩ := ideal.exists_pow_le_of_le_radical_of_fg _ _ J'.2
+      (is_noetherian_def.mp hR _),
+    refine ⟨costructured_arrow.mk _⟩,
+    exact (op k),
+    exact ⟨⟨hk⟩⟩ },
+  end }
+
+section
+/-
+Running into a universe issue on the next definition:
+using {R : Type u} as above causes an error.
+-/
+
+/-- Local cohomology (defined in terms of powers of `J`) agrees with local
+cohomology computed over all ideals with radical containing `J`. -/
+def iso_self_le_radical
+  {R : Type} [comm_ring R] (J : ideal R) [is_noetherian R R] (i : ℕ) :
```

Comment:
Awkward universe issue here (try changing `{R : Type}` to `{R : Type u}`).

Reply:
I pushed [9ae9af9](https://github.com/leanprover-community/mathlib/pull/19105/commits/9ae9af94644101bced7a40d2c866a89ffc22dcc8) for this.